### PR TITLE
480 collect x-api-client header

### LINKED ISSUE #480: Collect client Header Id in applicationinsights/appmetrics
https://blogs.msdn.microsoft.com/stuartleeks/2016/11/03/including-headers-for-request-events-with-application-insights/

### PR DESCRIPTION
**Pull request by Yshayy**
closes #480

### CODE REVIEW

File: addons/ApplicationInsights/Tweek.Addons.ApplicationInsights/ApplicationInsightAddon.cs [hunk 1]
```diff
@@ -1,15 +1,38 @@
...
+using Microsoft.ApplicationInsights.Extensibility;
 using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Tweek.ApiService.Addons;
 
 namespace Tweek.Addons.ApplicationInsights
 {
+    public class ApiClientTelemetryInitializer : ITelemetryInitializer
```
Comment (asafkotzer) [inline addons/ApplicationInsights/Tweek.Addons.ApplicationInsights/ApplicationInsightAddon.cs]: Why not in a separate file?

File: addons/ApplicationInsights/Tweek.Addons.ApplicationInsights/ApplicationInsightAddon.cs [hunk 2]
```diff
@@ -1,15 +1,38 @@
...
+        IHttpContextAccessor httpContextAccessor;
+
+        public ApiClientTelemetryInitializer(IHttpContextAccessor httpContextAccessor)
+        {
+            this.httpContextAccessor = httpContextAccessor;
+        }
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (httpContextAccessor.HttpContext == null) return;
+            if (httpContextAccessor.HttpContext.Request.Headers.TryGetValue("x-api-client", out var values)){
```
Comment (asafkotzer) [inline addons/ApplicationInsights/Tweek.Addons.ApplicationInsights/ApplicationInsightAddon.cs]: Request exists for sure? Couldn't find anything in docs.

File: addons/ApplicationInsights/Tweek.Addons.ApplicationInsights/ApplicationInsightAddon.cs [hunk 3]
```diff
@@ -1,15 +1,38 @@
...
+                telemetry.Context.Properties["x-api-client"] = values.ToString();
```
Comment (asafkotzer) [inline addons/ApplicationInsights/Tweek.Addons.ApplicationInsights/ApplicationInsightAddon.cs]: Extract a const for ```x-api-client```?

File: deployments/dev/docker-compose.vs.debug.yml [hunk 4]
```diff
@@ -8,10 +8,7 @@ services:
       dockerfile: Dockerfile
       args:
         source: ${DOCKER_BUILD_SOURCE}
-    depends_on:
```
Comment (asafkotzer) [inline deployments/dev/docker-compose.vs.debug.yml]: Why are we removing Graphite?

File: addons/ApplicationInsights/Tweek.Addons.ApplicationInsights/ApplicationInsightAddon.cs (hunk 2 above)
Comment (Yshayy) [inline addons/ApplicationInsights/Tweek.Addons.ApplicationInsights/ApplicationInsightAddon.cs]: I think HttpContext is created per request, so I don't think Request can be null

File: deployments/dev/docker-compose.vs.debug.yml (hunk 4 above)
Comment (Yshayy) [inline deployments/dev/docker-compose.vs.debug.yml]: Graphite was only for tesing the addon we've developed, there's no need to run it in order to run/debug Tweek locally

**Pull request merged**